pkg/util/config/config.go

```diff
@@ -51,16 +51,42 @@ type Resources struct {
 	ShmVolume               *bool               `name:"enable_shm_volume" default:"true"`
 }
 
+type InfrastructureRole struct {
+	// Name of a secret which describes the role, and optionally name of a
+	// configmap with an extra information
+	Secret spec.NamespacedName
```

Comment:
I wonder if this should also be called `SecretName` to be clear it's just a reference. What is the common way in K8s to name reference to other resources?